packages/app/prisma-utils/src/cockroachdbError.ts

```diff
@@ -0,0 +1,25 @@
+import type { PrismaClientKnownRequestError } from '@prisma/client/runtime/library'
+
+/**
+ * https://www.cockroachlabs.com/docs/stable/transaction-retry-error-reference#:~:text=To%20indicate%20that%20a%20transaction,the%20string%20%22restart%20transaction%22%20.
+ *
+ * All transaction retry errors use the SQLSTATE error code 40001, and emit error messages with the string restart transaction. Further, each error includes a specific error code to assist with targeted troubleshooting
+ */
+const COCKROACHDB_RETRY_TRANSACTION_CODE = '40001'
+const COCKROACHDB_RETRY_TRANSACTION_MESSAGE = 'restart transaction'
+
+/**
+ * Check if the error is a CockroachDB transaction retry error
+ *
+ * @param error
+ */
+export const isCockroachDBRetryTransaction = (error: PrismaClientKnownRequestError): boolean => {
+	const meta = error.meta
+	if (!meta) return false
+
+	return (
+		meta.code === COCKROACHDB_RETRY_TRANSACTION_CODE &&
+		typeof meta.message === 'string' &&
+		meta.message.includes(COCKROACHDB_RETRY_TRANSACTION_MESSAGE)
```

Comment:
is this part necessary? coupling to specific messages is not very reliable. we can't rely on code alone?

Reply:
We can yes, but I wanted to be as explicit as possible, this is a CockroachDB-specific error and we may want to use this library with other DBs and not sure if we can have the same code with different meanings

About that coupling to specific messages is not very reliable, I completely agree, but on doc, they explicitly say that the message should contain that part so I would say it is fine.

But I have no strong opinion, if you prefer to remove that check I am more than happy to do it

Reply:
fair point that this is a cockroach-specific error. can we do a more reliable check? does prisma client expose the used driver name?

Reply:
That was the first idea, but I couldn't find it exposed 😓

Reply:
Main thing that concerns me is that we don't have a real integration test throwing this error, so if CockroachDB ever decides to change the message, no tests will be failing.
what do you think about passing a db driver enum as a part of PrismaTransactionOptions?

Reply:
Fair point and good idea! yeah, will include the DB driver as part of the options.

I think I will make it mandatory which means a major release, so in the future when we add more drivers we don't need to migrate :D

Reply:
sounds good! we can also default to CRDB to retain backwards compatibility :D

Reply:
yeah, finally decided to use it as default, and later when we need to add more Db drivers we can see if make sense to make it mandatory